components/ThoughtList/index.js

```diff
@@ -1,158 +1,202 @@
-    addThought = (myThoughtList, index) => {
-        let obj = {}
-        obj.thought = this.state.thought
+    setInitialList(props.thoughtsList)
 
-        myThoughtList[index] = obj
-        if (index == myThoughtList.length - 1) {
-            myThoughtList.push({})
-        }
-        this.setState({
-           
-        }, () => {
-            console.log(this.state.thoughtsList)
-        })
+    const occurrences = tempList.filter((val) => {
+      return val === thought;
+    }).length;
+    if (occurrences > 1) {
+      setUpdatedCount(props.thoughtsList, thought, occurrences)
     }
+  }
 
-    /// <summary>
-    /// TODO : To create a subThought list in a Thought..
-    /// </summary>
-    /// <param name="myThoughtList">thought list in which data is to be added</param>
-    /// <param name="index">To insert thought at particular postion</param>
-    addSubThought = (myThoughtList, index) => {
-        console.log('long presss')
-        let obj = {}
-        if (myThoughtList[index].hasOwnProperty("subList")) {
-            myThoughtList[index].subList.push(obj)
-        }
-        else {
-            myThoughtList[index].subList = [obj]
-            myThoughtList[index].isOpen = true
+  /// <summary>
+  /// TODO : To convert the nested thought list to single list..
+  /// </summary>
+  /// <param name="myList">thought list or subthought list</param>
+  const setInitialList = (myList) => {
+    myList.forEach((item) => {
+      if (item.hasOwnProperty('thought')) {
+        tempList.push(item.thought)
+      }
+      if (item.hasOwnProperty('subList')) {
+        setInitialList(item.subList)
+      }
+    })
+  }
+
+  /// <summary>
+  /// TODO : Update the list with count of each thought..
+  /// </summary>
+  /// <param name="myList">thought list or subthought list</param>
+  /// <param name="thought">thought to be added</param>
+  /// <param name="occurrences">total ocurrences of the thought in thoughtList</param>
+  const setUpdatedCount = (myList, thought, occurrences) => {
+    myList.forEach((item) => {
+      if (item.hasOwnProperty('thought')) {
+        if (item.thought == thought) {
+          item.count = occurrences
+          const newList = [...props.thoughtsList]
+          props.addNewThought(newList)
         }
-        this.setState({})
+      }
+      if (item.hasOwnProperty('subList')) {
+        setUpdatedCount(item.subList, thought, occurrences)
+      }
+    })
+  }
+
+  /// <summary>
+  /// TODO : To create a subThought list in a Thought..
+  /// </summary>
+  /// <param name="myThoughtList">thought list in which data is to be added</param>
+  /// <param name="index">To insert thought at particular postion</param>
+  const addSubThought = (myThoughtList, index) => {
+    const obj = {}
+    if (myThoughtList[index].hasOwnProperty("subList")) {
+      myThoughtList[index].subList.push(obj)
+    }
+    else {
+      myThoughtList[index].subList = [obj]
+      myThoughtList[index].isOpen = true
```

Comment:
There are a lot of mutations in this code, which really goes against best the principles of this project. I'm concerned that the functional programming style is something you're super familiar with. Is this something you're willing to learn, or do you think it's going to be a problem to implement things in a functional style rather than an object-oriented style?